identity/identity.go

```diff
@@ -103,66 +64,115 @@ func GetIdentityHeader(ctx context.Context) string {
 	return ""
 }
 
-func checkHeader(id *XRHID, w http.ResponseWriter) error {
+var (
+	ErrMissingIdentity      = errors.New("missing x-rh-identity header")
+	ErrDecodeIdentity       = errors.New("unable to b64 decode x-rh-identity header")
+	ErrInvalidOrgIdIdentity = errors.New("x-rh-identity header has an invalid or missing org_id")
+	ErrMissingIdentityType  = errors.New("x-rh-identity header is missing type")
+)
 
+// CheckBasePolicy performs semantic identity check and returns nil for valid values or
+// an error instead.
+func CheckBasePolicy(id *XRHID) error {
 	if (id.Identity.Type == "Associate" || id.Identity.Type == "X509") && id.Identity.AccountNumber == "" {
 		return nil
 	}
 
 	if id.Identity.OrgID == "" && id.Identity.Internal.OrgID == "" {
-		return doError(w, 400, "x-rh-identity header has an invalid or missing org_id")
+		return ErrInvalidOrgIdIdentity
 	}
 
 	if id.Identity.Type == "" {
-		return doError(w, 400, "x-rh-identity header is missing type")
+		return ErrMissingIdentityType
 	}
 
 	return nil
 }
 
-// EnforceIdentity extracts the X-Rh-Identity header and places the contents into the
-// request context.  If the Identity is invalid, the request will be aborted.
-func EnforceIdentity(next http.Handler) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		rawHeaders := r.Header["X-Rh-Identity"]
+// If org_id is not defined at the top level, use the internal one. See:
+// https://issues.redhat.com/browse/RHCLOUD-17717
+func topLevelOrgIDFallback(identity *XRHID) {
```

Comment:
This function is only called once, inside `DecodeIdentity`. Is there any
particular reason why it needs to exist outside that function as a separate one?